adam_core/propagator/propagator.py

```diff
@@ -1,18 +1,23 @@
 import concurrent.futures
 import logging
 from abc import ABC, abstractmethod
-from typing import Optional
+from typing import Optional, Union
 
 from astropy.time import Time
 from quivr.concat import concatenate
 
-from ..orbits import Orbits
+from ..orbits.orbits import Orbits
+from ..orbits.variants import VariantOrbits
 from .utils import _iterate_chunks, sort_propagated_orbits
 
 logger = logging.getLogger(__name__)
 
+OrbitType = Union[Orbits, VariantOrbits]
 
-def propagation_worker(orbits: Orbits, times: Time, propagator: "Propagator") -> Orbits:
+
+def propagation_worker(
+    orbits: OrbitType, times: Time, propagator: "Propagator"
```

Comment:
Do we need to implement memory sharing / pass by reference here to avoid serialization costs?

Reply:
Yes, this is where that would come in and I think this is something that is best done upstream in quivr-land. @spenczar did some early testing of passing shared memory references but it hasn't been a priority. This could actually be something nice to get implemented sooner rather than later considering that our work on THOR is ramping up?